manim/mobject/svg/style_utils.py

```diff
@@ -0,0 +1,146 @@
+"""Utility functions for parsing SVG styles."""
+
+
+__all__ = ["cascade_element_style", "parse_style", "parse_color_string"]
+
+from xml.dom.minidom import Element as MinidomElement
+from colour import web2hex
+from ...utils.color import rgb_to_hex
+
+from typing import Dict
+
+
+SUPPORTED_STYLING_ATTRIBUTES = [
+    "fill",
+    "stroke",
+    "style",
+    "fill-opacity",
+    "stroke-opacity",
+]
+
+
+def cascade_element_style(
+    element: MinidomElement, inherited: Dict[str, str]
+) -> Dict[str, str]:
+    """Collect the element's style attributes based upon both its inheritance and its own attributes.
+
+    SVG uses cascading element styles. A closer ancestor's style takes precedence over a more distant ancestor's
+    style. In order to correctly calculate the styles, the attributes are passed down through the inheritance tree,
+    updating where necessary.
+
+    Note that this method only copies the values and does not parse them. See :meth:`parse_color_string` for converting
+    from SVG attributes to manim keyword arguments.
+
+    Parameters
+    ----------
+    element : :class:`MinidomElement`
+        Element of the SVG parse tree
+
+    inherited : :class:`dict`
+        Dictionary of SVG attributes inherited from the parent element.
+
+    Returns
+    -------
+    :class:`dict`
+        Dictionary mapping svg attributes to values with `element`'s values overriding inherited values.
+    """
+
+    style = inherited.copy()
+
+    for attr in SUPPORTED_STYLING_ATTRIBUTES:
+        entry = element.getAttribute(attr)
+        if entry:
+            style[attr] = entry
+
+    return style
+
+
+def parse_color_string(color_spec: str) -> str:
+    """Handle the SVG-specific color strings and convert them to HTML #rrggbb format.
+
+    Parameters
+    ----------
+    color_spec : :class:`str`
+        String in any web-compatible format
+
+    Returns
+    -------
+    :class:`str`
+        Hexadecimal color string in the format `#rrggbb`
+    """
+
+    if color_spec[0:3] == "rgb":
+        # these are only in integer form, but the Colour module wants them in floats.
+        splits = color_spec[4:-1].split(",")
+        if splits[0][-1] == "%":
+            # if the last character of the first number is a percentage,
+            # then interpret the number as a percentage
+            parsed_rgbs = [int(i[:-1]) / 100.0 for i in splits]
```

Comment:
Hello @MrMallIronmaker 

It would be safe to use float(i[:-1]) instead of int.

https://www.w3.org/TR/SVGColor12/#sRGBcolor

I tried an SVG (generated by Tikz) and it failed because of casting by int.

Locally when I changed the casting to float then it worked out. In that SVG file, there were mostly int-type entries and the casting by float tolerated them as well. So I believe it should be okay to use float as a generic cast.

Thank you for this great work.

Eagerly waiting for your pull request to be accepted :).

Regards
Kapil

Reply:
You are 100% right, this is a good fix.

Reply:
fixed in 898b46c